check_apt: add option to list package names

mafgh opened the pull request:
Updated version of #1430:

List names of pending package upgrades:

```
plugins/check_apt --input-file /tmp/foo -l
APT CRITICAL: 5 packages available for upgrade (2 critical updates) postgresql-client-common postgresql-client-9.1 postgresql-common postgresql-9.1 postgresql. |available_upgrades=5;;;0 critical_updates=2;;;0
```
Input file contents:
```
Inst postgresql-client-common (134wheezy4 Debian:7.11/oldstable [all])
Inst postgresql-client-9.1 (9.1.23-0+deb7u1 Debian-Security:7.0/oldstable [i386])
Inst postgresql-common (134wheezy4 Debian:7.11/oldstable [all])
Inst postgresql-9.1 (9.1.23-0+deb7u1 Debian-Security:7.0/oldstable [i386])
Inst postgresql (9.1+134wheezy4 Debian:7.11/oldstable [all])
Conf postgresql-client-common (134wheezy4 Debian:7.11/oldstable [all])
Conf postgresql-client-9.1 (9.1.23-0+deb7u1 Debian-Security:7.0/oldstable [i386])
Conf postgresql-common (134wheezy4 Debian:7.11/oldstable [all])
Conf postgresql-9.1 (9.1.23-0+deb7u1 Debian-Security:7.0/oldstable [i386])

```

waja commented:
This seems to be a possible fix for #1256. An alternative might be #1427.

waja commented:
I would prefer #1427 over this one. But what do I know. 😄

sni commented on plugins/check_apt.c:
```diff
@@ -148,22 +153,23 @@ int process_arguments (int argc, char **argv) {
 		case 'h':
 			print_help();
-			exit(STATE_UNKNOWN);
+			exit(STATE_OK);
```
Please never do that, this might lead to plugins which only print the help or version and you will never notice because they exit OK. See https://www.monitoring-plugins.org/doc/guidelines.html#AEN78

mafgh replied:
thanks, good catch.

sni commented on plugins/check_apt.c:
```diff
@@ -148,22 +153,23 @@ int process_arguments (int argc, char **argv) {
 		case 'V':
 			print_revision(progname, NP_VERSION);
-			exit(STATE_UNKNOWN);
+			exit(STATE_OK);
```
same here

mafgh commented:
I implemented @sni's changes and rebased onto master.

Please note, meanwhile master introduced the --only-critical option with bf18dc4, however my change will still list all pending updates. I am not sure if we should filter out non security updates when --only-critical is set.

waja commented:
> I am not sure if we should filter out non security updates when --only-critical is set.

I think it would make sense to filter them, when this option is set. At least I think this is what users would expect.
